cylc/flow/scripts/install.py

```diff
@@ -113,6 +113,44 @@ def get_option_parser():
         default=False,
         dest="no_symlinks")
 
+    # If cylc-rose plugin is available ad the --option/-O config
+    try:
+        __import__('cylc.rose')
+        parser.add_option(
+            "--opt-conf-key", "-O",
+            help=(
+                "Use optional Rose Config Setting"
+                "(If Cylc-Rose is installed)"
+            ),
+            action="append",
+            default=[],
+            dest="opt_conf_keys"
+        )
+        parser.add_option(
+            "--define", '-D',
+            help=(
+                "Each of these overrides the `[SECTION]KEY` setting in a "
+                "`rose-suite.conf` file. "
+                "Can be used to disable a setting using the syntax "
+                "`--define=[SECTION]!KEY` or even `--define=[!SECTION]`."
+            ),
+            action="append",
+            default=[],
+            dest="defines"
+        )
+        parser.add_option(
+            "--define-suite", "--define-flow", '-S',
+            help=(
+                "As `--define`, but with an implicit `[SECTION]` for "
```

Comment:
Would functional tests be worthwhile here? I understand there are cylc-rose tests, although worth preserving business logic for integration? I can comment out all these options, (pip install) and have the cylc-rose tests passing. Unless I am doing something wrong (which is very possible).

Reply:
Yes.

Reply:
I've now created a functional test, but of course it's dependent on the Cylc-flow installed being on the right version so that now fails on GH actions... 
🙄